File: compiled/Junction.js
/// <reference path="Utils/Set.ts" />
/// <reference path="Road.ts" />
/// <reference path="Utils/Point.ts" />
var Junction = (function () {
    function Junction(id, loc) {
        this.id = id;
        this.location = loc;
        this.roads = new Set();
    }
    Junction.prototype.registerRoad = function (road) {
        this.roads.add(road);
    };
    Junction.prototype.hadRoad = function (road) {
        var list = this.roads;
        for (var i = 0; i < list.size(); i++) {
            if (list.getItem(i).equals(road)) {
                return true;
            }
        }
        return false;
    };
    Junction.prototype.hasRoadWithID = function (id) {
        var list = this.roads;
        for (var i = 0; i < list.size(); i++) {
            if (list.getItem(i).id == id) {
                return true;
            }
        }
        return false;
    };
    Junction.prototype.hasRoad = function (road) {
        return this.roads.contains(road);
    };
    Junction.prototype.equals = function (junction) {
        if (junction == null) {
            return false;
        }
        else if (this.id == junction.id) {
            return true;
        }
        return false;
    };
    return Junction;
}());

File: src/Junction.py
class Junction(object):
    "Defines a junction in the model!"
    def __init__(self, id, loc):
        self.id = id
        self.location = loc
        self.roads = set()

    def register_road(self, road):
        self.roads.add(road)
    
    def has_road(self, road):
        return road in self.roads
    
    def has_road_with_id(self, id):
        for road in self.roads:
            if road.id == id:
                return True
        return False
    
    def __eq__(self, junction):
        return self.id == junction.id